db-cleaner/src/cleaner.rs

```diff
@@ -0,0 +1,180 @@
+        credentials,
+        Region::EuWest1
+    );
+
+    let path: String = "/notifications/".to_owned() + &stage + "/workers";
+
+    let req = GetParametersByPathRequest {
+        max_results: None,
+        next_token: None,
+        parameter_filters: None,
+        path: path.to_string(),
+        recursive: Some(true),
+        with_decryption: Some(true),
+
+    };
+
+    ssm_client.get_parameters_by_path(req).sync()
+        .map_err(|error| {
+            error!("Error while fetching parameter: {}", error);
+            "Unable to fetch configuration".to_string()
+        })
+        .and_then(|result| {
+            result.parameters.ok_or("No Parameter Found".to_string())
+        })
+        .map(|parameters| {
+            for parameter in parameters {
+                if parameter.name.is_some() && parameter.value.is_some() {
+                    let prefix_length = path.len() + 1; // +1 for the extra slash
+                    let name = parameter.name.unwrap()[prefix_length..].to_string();
+                    config.insert(name, parameter.value.unwrap());
+                }
+            }
+            config
+        })
+        .and_then(|config| {
+            if config.is_empty() {
+                error!("No configuration loaded");
+                Err("Unable to fetch configuration".to_string())
+            } else { Ok(config) }
+        })
+}
+
+fn config_to_connection_parameter(config: HashMap<String, String>) -> Result<ConnectionParameters, String> {
+    config.get("cleaner.registration.db.url").and_then(|jdbc_url| {
+        Some(ConnectionParameters {
+            jdbc_url: jdbc_url.to_owned(),
+            user: config.get("cleaner.registration.db.user")?.to_owned(),
+            password: config.get("cleaner.registration.db.password")?.to_owned(),
+        })
+    }).ok_or("Unable to read the url, user or password from the configuration".to_string())
+}
+
+fn create_connection_url(connection_parameters: ConnectionParameters, local: bool) -> Result<String, String> {
+    let url = connection_parameters.jdbc_url.replace("jdbc:", "");
+    match Url::parse(&url) {
+        Ok(mut url) => {
+            url.set_username(&connection_parameters.user);
+            url.set_password(Some(&connection_parameters.password));
+            url.set_query(None);
+
+            if local { url.set_host(Some("localhost")); };
+
+            Ok(url.as_str().to_string())
+        }
+        Err(e) => Err(e.to_string())
+    }
+
+}
+
+pub fn delete_outdated_rows(local_context: bool) -> Result<u64, String> {
+    let request = "delete from registrations.registrations where lastmodified <= now() - interval '120 days';";
+
+    get_credentials()
+        .and_then(fetch_config)
+        .and_then(config_to_connection_parameter)
+        .and_then(|params| create_connection_url(params, local_context))
+        .and_then(|connection_url| Connection::connect(connection_url, TlsMode::None).map_err(|e| e.to_string()))
```

Comment:
Ignore this, just noticed your comment in the description. I feel like TLS is worth adding, especially as this may be used a reference for future projects so let's see how the issue goes.